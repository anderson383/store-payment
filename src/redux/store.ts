import { combineReducers, configureStore } from '@reduxjs/toolkit';
import paymentReducer from "./payment/payment.root";

export const rootReducer = combineReducers({
  paymentReducer: paymentReducer
});

type RootState = ReturnType<typeof rootReducer>;

const store = configureStore<RootState>({
  reducer: rootReducer
})

export type RootStateType = ReturnType<typeof store.getState>

export default store;
