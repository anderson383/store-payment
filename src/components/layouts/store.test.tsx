
import {
  render,
  screen
} from '@testing-library/react';
import { act } from 'react';
import { StoreLayout } from './store.layout';

describe('Tests in Store oayout component', () => {
  const setup = () => {
    return render(
      <StoreLayout>
        <div>Contenido renderizado correctamente</div>
      </StoreLayout>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });
});
