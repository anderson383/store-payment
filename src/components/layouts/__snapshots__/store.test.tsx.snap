// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Store oayout component Should render the component 1`] = `
<DocumentFragment>
  <div
    class="storeLayout"
  >
    <header
      class="header"
    >
      <div>
        <h1>
          STORE ASTETIC
        </h1>
      </div>
    </header>
    <main>
      <div>
        Contenido renderizado correctamente
      </div>
    </main>
  </div>
</DocumentFragment>
`;
